tree/ntuple/v7/src/RPageSinkBuf.cxx

```diff
@@ -37,11 +37,47 @@ void ROOT::Experimental::Detail::RPageSinkBuf::CreateImpl(const RNTupleModel &mo
                                                           unsigned char * /* serializedHeader */,
                                                           std::uint32_t /* length */)
 {
-   fBufferedColumns.resize(fDescriptorBuilder.GetDescriptor().GetNPhysicalColumns());
    fInnerModel = model.Clone();
    fInnerSink->Create(*fInnerModel);
 }
 
+void ROOT::Experimental::Detail::RPageSinkBuf::UpdateDescriptor(const RNTupleModelChangeset &changeset)
+{
+   RPageSink::UpdateDescriptor(changeset);
+   bool isIncremental = !fBufferedColumns.empty();
+   fBufferedColumns.resize(fDescriptorBuilder.GetDescriptor().GetNPhysicalColumns());
+   if (!isIncremental)
+      return;
+
+   // The buffered page sink maintains a copy of the RNTupleModel for the inner sink; replicate the changes there
```

Comment:
Let's add a TODO. We should be able, in general, to simplify the buffered sink. But this is for another PR.